### ISSUE TITLE
Selection mode should not activate when selection is single token

### ISSUE THREAD

**Issue by Willyham** [Kraoz/atom-auto-replace-in-selection#1]
I often double-click to select a given token, then copy, hit ctrl+f and paste that token into the search bar. I would argue that this is a common case and it is unlikely that anybody would want to search 'in selection' within a single token.

My recommendation would be to parse tokenize the selection (on spaces and or common delimiters) and if there is only one result, do not activate the mode.

Thoughts?

**Comment by cornelius-dol** [comment]
Agreed. TextPad handles automatic search in selection based on whether at least one line break is selected. Multi-line search in selection; within a single line search file. I found this to work really well. But single-token or not would hit 90% of my use cases too (I sometime do select a short phrase to search for it in the rest of the file; if I had to cancel the selection first I could live with that).
